tutorials/generating-synthetic-data/synthetic-data-generation-6.md

```diff
@@ -1,225 +1,195 @@
-| 0.0255                        | 3 minutes 20 seconds    | 0.2808     |
-| 0.05                          | 8 minutes 17 seconds    | 0.6983     |
-
-**Not sure why the values for 0.05 times step are larger.** Maybe the simulation is erroneous and all particles clump together.
+VMs also allow you to configure the number of virtual CPUs (vCPUs), ranging 
+from as few as 4 to several dozen, or even hundreds, on machine families 
+like `c2d` and `c3d`. While increasing the number of vCPUs generally speeds 
+up simulations, scaling is rarely linear. 
+
+There’s often a “sweet spot” where performance gains flatten out as parallelization 
+efficiency decreases. Beyond this point, using more vCPUs may result in 
+diminishing returns, where costs rise faster than the benefits of additional 
+speed. We’ll explore this tradeoff in detail in the benchmark results later.
+
+## Defining the Benchmark Parameters
+
+To benchmark the performance and cost-efficiency of different machine types, we 
+need to evaluate computation times and scalability without unnecessary overhead.
+
+
+
+First, we need to download the necessary input files. This step is needed because the benchmark requires specific data to 
+simulate the **WaterCube** environment. The code snippet uses the ```download_from_url``` method to download and unzip the 
+required files directly into your local directory. You can skip this step if you already have the input files in your local 
+directory.
+
+```python
+import inductiva
+
+inductiva.utils.files.download_from_url(
+    url="https://tutorials.inductiva.ai/_static/generating-synthetic-data/"
+        "splishsplash-base-dir.zip",
+    unzip=True)
+```
+
+Once the input files are downloaded, it's a good idea to upload them to a remote storage location. This step prevents you 
+from having to re-upload the input files every time you execute a run during the benchmark, which can save time (e.g., 
+reduces setup time) when you're executing multiple runs (i.e., simulations). In other words, uploading the files ensures that 
+they can be accessed across different benchmark runs and it also helps manage resources more efficiently.
+
+```python
+import inductiva
+
+inductiva.storage.upload(local_path="splishsplash-base-dir",
+                         remote_dir="splishsplash-input-dir")
+```
+
+Now comes the core of the benchmarking process: configuring the benchmark with the desired settings and running it across 
+multiple machine types. In this step, you will define the benchmark parameters, such as the simulator used, the simulation 
+configuration file, and the remote assets (i.e., the input files you just uploaded). The benchmark is configured to run 
+across a range of machine families, from the less powerful ```c2-standard``` machine types to the more powerful ```c3``` 
+machine types, each with varying numbers of vCPUs. Each run is repeated twice to ensure reliable and consistent results.
+
```

Comment:
Do we need to explain that splishsplash automatically takes advantage of an increased number of vCPUs, that is, we don't have to explicitly set any level of paralelization ourselves?

Reply:
Yes that's an important detail -- the configuration is much easier this way. I'm going to add that.

Reply:
"An important point to note is that the **SPlisHSPlasH** simulator automatically handles parallelization, taking full advantage of the increased number of vCPUs on a machine without requiring additional configuration. This is not true for all simulators, such as **Reef3D**, which requires updating certain parameters in the configuration file."